src/ucp/core/ucp_am.c

```diff
@@ -813,7 +878,52 @@ UCS_PROFILE_FUNC(ucs_status_ptr_t, ucp_am_recv_data_nbx,
                  ucp_worker_h worker, void *data_desc, void *buffer,
                  size_t count, const ucp_request_param_t *param)
 {
-    return UCS_STATUS_PTR(UCS_ERR_NOT_IMPLEMENTED);
+    ucp_am_rndv_rts_hdr_t *rts = data_desc;
+    ucs_status_ptr_t ret;
+    ucp_request_t *req;
+    ucp_datatype_t datatype;
+
+    UCP_CONTEXT_CHECK_FEATURE_FLAGS(worker->context, UCP_FEATURE_AM,
+                                    return UCS_STATUS_PTR(UCS_ERR_INVALID_PARAM));
+    UCP_WORKER_THREAD_CS_ENTER_CONDITIONAL(worker);
+
+    ucs_assert(rts->super.flags & UCP_RNDV_RTS_FLAG_AM);
+
+    req = ucp_request_get_param(worker, param,
+                                {ret = UCS_STATUS_PTR(UCS_ERR_NO_MEMORY);
+                                 goto out;});
+
```

Comment:
so this means we will have both send request and receive request for rndv? do we really need both?

Reply:
do you mean rndv request by send request? This is the flow we use for tag matching as well

Reply:
but for tag matching we must create receive request because the receive is started by API call and it could be "expected". 
for data fetch i don't think we really need a receive request

Reply:
but it is `ucp_am_recv_data_nbx` API call, which is supposed to return a request

Reply:
can we return the send request?